src/sapling/sapling_operation.cpp

```diff
@@ -65,6 +65,7 @@ TxValues calculateTarget(const std::vector<SendManyRecipient>& recipients, const
 
 OperationResult SaplingOperation::build()
 {
+    AssertLockHeld(pwalletMain->cs_wallet);
```

Comment:
This isn't needed. If we have to lock `cs_wallet` due the new changes (like the nullifier set), it should be done inside this method.